build-tools/scripts/jenkinsfile

```diff
@@ -0,0 +1,484 @@
+                        csvSeries: [[
+                            displayTableFlag: true, file: 'TestResult-Xamarin.Forms_Tests-values.csv', inclusionFlag: 'OFF'
+                        ]],
+                        group: 'Tests size', logarithmic: true, style: 'line', yaxis: 'ms'
+                    )
+
+                    plot(
+                        title: 'Hello World',
+                        csvFileName: 'plot-hello-world-build-times.csv',
+                        csvSeries: [[
+                            displayTableFlag: true, file: 'TestResult-Timing-HelloWorld.csv', inclusionFlag: 'OFF'
+                        ]],
+                        group: 'Build times', logarithmic: true, style: 'line', yaxis: 'ms'
+                    )
+                    
+                    plot(
+                        title: 'Xamarin.Forms',
+                        csvFileName: 'plot-xamarin-forms-integration-build-times.csv',
+                        csvSeries: [[
+                            displayTableFlag: true, file: 'TestResult-Timing-Xamarin.Forms-Integration.csv', inclusionFlag: 'OFF'
+                        ]],
+                        group: 'Build times', logarithmic: true, style: 'line', yaxis: 'ms'
+                    )
+                }
+
+                echo "Stage result: Plot build & test metrics: ${currentBuild.currentResult}"
+            }
+        }
+
+        stage('Upload test results') {
+            options {
+                timeout(time: 5, unit: 'MINUTES')       // Typically takes under 1 minute to upload test results
+            }
+
+            steps {
+                script {
+                    try {
+                        xunit thresholds: [
+                                failed(unstableNewThreshold: '0', unstableThreshold: '0'),
+                                skipped()                                                       // Note: Empty threshold settings per settings in the xamarin-android freestyle build are not permitted here
+                            ],
+                            tools: [
+                                NUnit2(deleteOutputFiles: true,
+                                failIfNotNew: true,
+                                pattern: 'xamarin-android/TestResult-*.xml',
+                                skipNoTestFiles: true,
+                                stopProcessingIfError: false)
+                            ]
+                    } catch (error) {
+                        echo "ERROR : Upload test results: Unexpected error: ${error}. Marking build as UNSTABLE"
+                        currentBuild.result = 'UNSTABLE'
+                    }
+                }
+
+                echo "Stage result: Upload test results: ${currentBuild.currentResult}"
+            }
+        }
+    }
+
+    post {
+        // Attempt to process & upload build results if the initial attempt to process and upload the logs failed
```

Comment:
Is this block likely to be useful?  If it failed the first time, is it really likely that trying again here will fix it?

Reply:
I should note with the move to the scripted pipeline, the post block was removed as it is a feature specific to the declarative pipeline). Based on the test builds that I've been running, I haven't seen failures (yet) that require retries.  If we run into any common failures due to connectivity issues going forward, I propose handling them via issues on a case-by-case basis.